packages/compass/src/app/components/home.tsx

```diff
@@ -198,13 +201,65 @@ function Home({
   });
 
   const onConnected = useCallback(
-    (connectionInfo: ConnectionInfo, dataService: DataService) => {
-      trackNewConnectionEvent(connectionInfo, dataService, logger, track);
+    (
+      connectionInfo: ConnectionInfo,
+      dataService: DataService,
+      activeAndInactiveConnectionsCount: { active: number; inactive: number }
+    ) => {
+      trackNewConnectionEvent(
+        connectionInfo,
+        dataService,
+        logger,
+        track,
+        activeAndInactiveConnectionsCount
+      );
       dispatch({ type: 'connected', connectionInfo: connectionInfo });
     },
     [logger, track]
   );
 
+  const onDisconnected = useCallback(
+    (
+      connectionInfo: ConnectionInfo | undefined,
+      activeAndInactiveConnectionsCount: { active: number; inactive: number }
+    ) => {
+      trackConnectionDisconnectedEvent(
+        connectionInfo,
+        track,
+        activeAndInactiveConnectionsCount
+      );
+    },
+    [track]
+  );
+
+  const onConnectionCreated = useCallback(
+    (
+      connectionInfo: ConnectionInfo | undefined,
+      activeAndInactiveConnectionsCount: { active: number; inactive: number }
+    ) => {
+      trackConnectionCreatedEvent(
+        connectionInfo,
+        track,
+        activeAndInactiveConnectionsCount
+      );
+    },
+    [track]
+  );
+
+  const onConnectionRemoved = useCallback(
+    (
+      connectionInfo: ConnectionInfo | undefined,
+      activeAndInactiveConnectionsCount: { active: number; inactive: number }
+    ) => {
+      trackConnectionRemovedEvent(
```

Comment:
Similar to my previous comment. I'm curious about what prompted this "callback" / "event-driven" style here.
 
It seems `onDisconnected`, `onConnectionCreated`, and `onConnectionRemoved` exclusively call the respective track functions and require adapting the calling point to surface additional information.

What is the benefit of this approach? From my perspective, this indirection might be obscuring the code's intention, making it harder to follow. I'm open to discuss this further of course, but i don't see much of an advantage to be doing this instead of just calling trackConnection<X>Event directly.

Reply:
We initially had https://jira.mongodb.org/browse/COMPASS-7740 opened for this, but it's assigned to me and closed. I don't really remember why it got resolved like that, but I'm guessing we decided that it will be just resolved with some other refactoring that still needs to happen?

Reply:
yeah, that's probably a good guess. Though we probably thought you were doing this refactoring @gribnoysup :)

Reply:
> desktop-only dependencies so had to stay away from the code shared with web

Isn't that addressed by providing a different implementation of services?